### PR TITLE
OFS-146: changes with saving contract contribution plan details 

### PR DESCRIPTION
**Pull request by sniedzielski**
This PR should be merged after https://github.com/openimis/openimis-be-contribution_plan_py/pull/22

In that PR I want to merge some fixes related to the changes on wiki page with approving contract (also valuation contract, submit, amend etc) and related them to the policies etc. This is necessary to work further on activatingPolicies when the payment is received (paid) etc.

### CODE REVIEW

File: contract/services.py [hunk 1]
```diff
@@ -417,10 +555,40 @@ def contract_valuation(self, contract_contribution_plan_details):
                 ccpd_record = model_to_dict(ccpd)
                 ccpd_record["calculated_amount"] = calculated_amount
                 if contract_contribution_plan_details["save"]:
-                    ccpd.save(self.user)
-                    uuid_string = f"{ccpd.id}"
-                    ccpd_record['id'], ccpd_record['uuid'] = (uuid_string, uuid_string)
-                ccpd_list.append(ccpd_record)
+                    from core import datetime, datetimedelta
```
Comment (delcroip) [inline contract/services.py:558]: please create another function

File: contract/services.py [hunk 2]
```diff
@@ -417,10 +555,40 @@ def contract_valuation(self, contract_contribution_plan_details):
...
+                    length = get_contribution_length(cpbd.contribution_plan)
+                    ccpd.date_valid_from = contract_details["contract_date_valid_from"]
```
Comment (delcroip) [inline contract/services.py:560]: add TODO: calculate the number of CCPD to create in order to cover the contract lenght

File: contract/services.py [hunk 3]
```diff
@@ -417,10 +555,40 @@ def contract_valuation(self, contract_contribution_plan_details):
...
+                    ccpd.date_valid_to = contract_details["contract_date_valid_from"] + datetimedelta(months=length)
+                    ccpd_results = self.create_ccpd(ccpd, contract_details["insuree_id"])
+                    ccpd_record = model_to_dict(ccpd)
+                    ccpd_record["calculated_amount"] = calculated_amount
+                    # case 1 - single contribution
+                    if len(ccpd_results) == 1:
```
Comment (delcroip) [inline contract/services.py:566]: add TODO: support more that 2 CCPD

**Review by delcroip — CHANGES_REQUESTED** [review]

**Comment by sniedzielski** [comment]
@delcroip I think that the most important issues are solved after changes related to the review

**Review by delcroip — APPROVED** [review]

**Pull request merged**